roles/splunk_deployer/tasks/main.yml

```diff
@@ -21,6 +21,18 @@
 - name: Flush restart handlers
   meta: flush_handlers
 
+- name: Wait for SHC to be ready 
+  shc_ready:
+    captain_url="{{ splunk.search_head_cluster_url }}"
+    shc_peers="{{ groups['splunk_search_head'] }}"
```

Comment:
The current requirement for having to specify the search head peers is a containerization anti-pattern, and really annoying as an operator. Ideally we could simply boot new search heads and have it self register with the cluster without every having to touch the captain. I realize that's out of scope for this PR though, but it does pose the question, should this be avoided as a requirement in hope of improving things in the future?

Correct me if I'm wrong, but we should be able to push the bundle without the cluster being fully bootstrapped, as long as the captain is up and ready. Then the captain can worry about replication to it's peers. Unless I'm missing something?

Reply:
Totally correct on the pattern, this exists so splunk-ansible can also support running outside of containers as well.  Long term we do plan on supporting compose "scale" commands with a single node, and k8 scale of a service.  That coding logic will likely be dealt with more on the container side than this implementation in splunk-ansible.

Reply:
That’s totally fair and something I didn’t consider. A suggestion would be to support an empty value for search heads and indexers, in conjunction with shc being set to true to trigger logic that causes non-captains to auto join the cluster (without dropping the rep factor 1, as currently does). Again though, out of scope.

Reply:
Regarding the Ansible task pattern, I would much prefer to have module just check once when it is executed and use repeat on the task level. This bring more clarity and control over how many times we want to repeat and wait checking on the shc.